### PR TITLE
Circle tests storybook

### PR DESCRIPTION
**Pull request by ErikOstlund**
Package: `lib-classifier`

## Describe your changes:
This PR is part of an effort to ensure all drawing tools have unit tests and a Storybook component.

This one adds unit tests and a Storybook component for the `circle` drawing tool.

<em>Helpful explanations that will make your reviewer happy:</em>
- What Zooniverse project should my reviewer use to review UX?
No specific projects are affected.

- What user actions should my reviewer step through to review this PR?
Checkout this branch and run `yarn test`

- Which storybook stories should be reviewed?
Run `yarn storybook` and go to: `localhost:6006`
In the `DRAWING TOOLS` section, you will see the Circle component.

- Are there plans for follow up PR’s to further fix this bug or develop this feature?
No

# Review Checklist

## General

- [ ] Are the tests passing locally and on Travis?
- [ ] Is the documentation up to date?

## Components
- [ ] Has a storybook story been created or updated?
- [ ] Is the component accessible? 
  - [ ] Can it be used with a screen reader? [BBC guide to testing with VoiceOver](https://bbc.github.io/accessibility-news-and-you/accessibility-and-testing-with-voiceover-os.html)
  - [ ] Can it be used from the keyboard? [WebAIM guide to keyboard testing](https://webaim.org/techniques/keyboard/#testing)
  - [ ] Is it passing accessibility checks in the storybook?

## Apps

- [ ] Does it work in all major browsers: Firefox, Chrome, Edge, Safari?
- [ ] Does it work on mobile?
- [ ] Can you `yarn panic && yarn bootstrap` or `docker-compose up --build` and app works as expected?

## Publishing

- [ ] Is the changelog updated?
- [ ] Are the dependencies updated for apps and libraries that are using the newly published library?

## Post-merging

- [ ] Did the app deploy to https://frontend.preview.zooniverse.org/projects/:project-name/:owner or https://frontend.preview.zooniverse.org/about?
- [ ] Is the new feature working or bug now fixed?
  - [ ] Is there a Talk or blog post written to announce the new feature(s)?
- [ ] Is the design working across browsers (Firefox, Chrome, Edge, Safari) and mobile?
  - [ ] Is this approved by our designer?
- [ ] Is this ready for production deployment?

### CODE REVIEW

File: packages/lib-classifier/src/plugins/drawingTools/components/Circle/Circle.js [hunk 1]
```diff
@@ -47,9 +47,27 @@ function Circle({ active, mark, onFinish, scale }) {
...
+    Circle Attributes:
+  */
+  mark: PropTypes.shape({
+    r: PropTypes.number,
+    x_center: PropTypes.number,
+    y_center: PropTypes.number
+  }).isRequired,
+
+  /**
+    Callback to reset the drawing canvas when creation of the rectangle is finished.
```
Comment (eatyourgreens) [inline packages/lib-classifier/src/plugins/drawingTools/components/Circle/Circle.js:64]:
```suggestion
    Callback to reset the drawing canvas when creation of the circle is finished.
```

File: packages/lib-classifier/src/plugins/drawingTools/components/Circle/Circle.spec.js [hunk 2]
```diff
@@ -1,27 +1,55 @@
 import React from 'react'
-import { render, screen } from '@testing-library/react'
+import { shallow, mount } from 'enzyme'
```
Comment (eatyourgreens) [inline packages/lib-classifier/src/plugins/drawingTools/components/Circle/Circle.spec.js:2]: I think reverting to Enzyme from React Testing Library might be a bad idea. Enzyme has held us up from updating React in the past, and 18 is a release candidate now.

File: packages/lib-classifier/src/plugins/drawingTools/components/Circle/Circle.spec.js [hunk 3]
```diff
@@ -1,27 +1,55 @@
...
+
+    const dragMove = wrapper
+      .find(DragHandle)
+      .find('[x=50][y=0][dragMove]')
+      .prop('dragMove')
+    expect(mark.x_center).to.equal(200)
+    expect(mark.y_center).to.equal(200)
+    expect(mark.r).to.equal(50)
+
+    console.log('mark.r: ', mark.r)
```
Comment (eatyourgreens) [inline packages/lib-classifier/src/plugins/drawingTools/components/Circle/Circle.spec.js:46]: Stray console log?

**Review by eatyourgreens — COMMENTED** [review]
The story looks good but I'm not at all sure about moving back to Enzyme, from React Testing Library. That seems like a bad idea, to me.

File: packages/lib-classifier/src/plugins/drawingTools/components/Circle/Circle.spec.js (hunk 2 above)
Comment (ErikOstlund) [reply packages/lib-classifier/src/plugins/drawingTools/components/Circle/Circle.spec.js:2]: I agree. My thought was to quick get a test out there so this is at least covered with a test for now, then come back to it at some point to convert to RTL... but we all know how that goes! ;) So, I will take the time to dive into RTL more and figure out how to convert this now.

File: packages/lib-classifier/src/plugins/drawingTools/components/Circle/Circle.spec.js (hunk 3 above)
Comment (ErikOstlund) [reply packages/lib-classifier/src/plugins/drawingTools/components/Circle/Circle.spec.js:46]: Yep - good catch.

**Comment by ErikOstlund** [comment]
Converting this to a Draft PR to convert unit tests to React Testing Library.

**Comment by ErikOstlund** [comment]
Figured out how to test this using React Testing Library.
Also added a `testid` prop to the `DragHandle` component so we can send unique test id names for drawing tools having more than one `DragHandle`.

e.g. The Rectangle uses the `DragHandle` component four times (one for each corner). We can send different test id names to each `DragHandle`:
`rect1-dragHandle`
`rect2-dragHandle`
`rect3-dragHandle`
`rect4-dragHandle`

**Review by eatyourgreens — APPROVED** [review]
Looks good to me 👍 .

**Pull request merged**